### PR TITLE
add CreateGender endpoint #2

### PR DESCRIPTION
**Pull request by VladSpitsin**

### CODE REVIEW

File: src/UserService.Models.Dto/Requests/User/Gender/CreateGenderRequest.cs [hunk 1]
```diff
@@ -0,0 +1,10 @@
+﻿using System;
+
+namespace LT.DigitalOffice.UserService.Models.Dto.Requests.User
+{
+  public class CreateGenderRequest
+  {
+    public Guid UserId { get; set; }
```
Comment (MariyaSin) [inline src/UserService.Models.Dto/Requests/User/Gender/CreateGenderRequest.cs:7]: you create new gender, for what userId here?

File: src/UserService.Business/Commands/Gender/CreateGenderCommand.cs [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+      _accessValidator = accessValidator;
+      _validator = validator;
+      _responseCreator = responseCreator;
+    }
+
+    public async Task<OperationResultResponse<Guid?>> ExecuteAsync(CreateGenderRequest request)
+    {
+      OperationResultResponse<Guid?> response = new();
+
+      if (request.UserId != _httpContextAccessor.HttpContext.GetUserId())
```
Comment (MariyaSin) [inline src/UserService.Business/Commands/Gender/CreateGenderCommand.cs:49]: no need do te check

File: src/UserService.Business/Commands/Gender/CreateGenderCommand.cs [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+      }
+
+      response.Body = await _genderRepository.CreateAsync(_mapper.Map(request));
+      response.Status = OperationResultStatusType.FullSuccess;
+
+      _httpContextAccessor.HttpContext.Response.StatusCode = (int)HttpStatusCode.Created;
+
+      if (response.Body == default)
+      {
+        return _responseCreator.CreateFailureResponse<Guid?>(HttpStatusCode.BadRequest);
```
Comment (MariyaSin) [inline src/UserService.Business/Commands/Gender/CreateGenderCommand.cs:66]: response = _responseCreator.CreateFailureResponse<Guid?>(HttpStatusCode.BadRequest);

File: src/UserService.Validation/Gender/CreateGenderRequestValidator.cs [hunk 4]
```diff
@@ -0,0 +1,18 @@
...
+
+namespace LT.DigitalOffice.UserService.Validation.Gender
+{
+  public class CreateGenderRequestValidator : AbstractValidator<CreateGenderRequest>, ICreateGenderRequestValidator
+  {
+    public CreateGenderRequestValidator(IGenderRepository genderRepository)
+    {
+      RuleFor(s => s.Name.Trim())
+        .NotEmpty().WithMessage("Gender must not be empty.")
+        .Must(name => !genderRepository.DoesGenderAlreadyExist(name))
```
Comment (MariyaSin) [inline src/UserService.Validation/Gender/CreateGenderRequestValidator.cs:14]: it must be async method and in the command you must await

File: src/UserService.Business/Commands/Gender/CreateGenderCommand.cs [hunk 5]
```diff
@@ -0,0 +1,67 @@
...
+      _genderRepository = genderRepository;
+      _validator = validator;
+      _responseCreator = responseCreator;
+    }
+
+    public async Task<OperationResultResponse<Guid?>> ExecuteAsync(CreateGenderRequest request)
+    {
+      OperationResultResponse<Guid?> response = new();
+
+      if (!_validator.ValidateCustom(request, out List<string> errors))
```
Comment (MariyaSin) [inline src/UserService.Business/Commands/Gender/CreateGenderCommand.cs:45]: add await

File: src/UserService.Business/Commands/Gender/CreateGenderCommand.cs [hunk 6]
```diff
@@ -0,0 +1,67 @@
...
+      response.Body = await _genderRepository.CreateAsync(_mapper.Map(request));
+      response.Status = OperationResultStatusType.FullSuccess;
+
+      _httpContextAccessor.HttpContext.Response.StatusCode = (int)HttpStatusCode.Created;
+
+      if (response.Body == default)
+      {
+        response = _responseCreator.CreateFailureResponse<Guid?>(HttpStatusCode.BadRequest);
+
+        return response;
```
Comment (MariyaSin) [inline src/UserService.Business/Commands/Gender/CreateGenderCommand.cs:61]: remove

File: src/UserService.Validation/Gender/CreateGenderRequestValidator.cs [hunk 7]
```diff
@@ -0,0 +1,18 @@
...
+
+namespace LT.DigitalOffice.UserService.Validation.Gender
+{
+  public class CreateGenderRequestValidator : AbstractValidator<CreateGenderRequest>, ICreateGenderRequestValidator
+  {
+    public CreateGenderRequestValidator(IGenderRepository genderRepository)
+    {
+      RuleFor(s => s.Name.Trim())
+        .NotEmpty().WithMessage("Gender must not be empty.")
+        .MustAsync(async (name, _) => await genderRepository.DoesGenderAlreadyExistAsync(name) != true)
```
Comment (MariyaSin) [inline src/UserService.Validation/Gender/CreateGenderRequestValidator.cs:14]: remove != true

File: src/UserService.Validation/Gender/CreateGenderRequestValidator.cs [hunk 8]
```diff
@@ -0,0 +1,18 @@
...
+using LT.DigitalOffice.UserService.Models.Dto.Requests.User;
+using LT.DigitalOffice.UserService.Validation.Gender.Interfaces;
+
+namespace LT.DigitalOffice.UserService.Validation.Gender
+{
+  public class CreateGenderRequestValidator : AbstractValidator<CreateGenderRequest>, ICreateGenderRequestValidator
+  {
+    public CreateGenderRequestValidator(IGenderRepository genderRepository)
+    {
+      RuleFor(s => s.Name.Trim())
```
Comment (MariyaSin) [inline src/UserService.Validation/Gender/CreateGenderRequestValidator.cs:12]: you get null ref if name will be null
s => s.Name?.Trim()

File: src/UserService/Program.cs [hunk 9]
```diff
@@ -5,55 +5,55 @@
...
+      }
+    }
+
+    public static IHostBuilder CreateHostBuilder(string[] args) =>
+      Host.CreateDefaultBuilder(args)
+        .UseSerilog()
+        .ConfigureWebHostDefaults(webBuilder =>
+        {
+          webBuilder.UseStartup<Startup>();
+        });
```
Comment (MariyaSin) [inline src/UserService/Program.cs:57]: update specification version in startup

File: src/UserService.Validation/Gender/CreateGenderRequestValidator.cs (hunk 8 above)
Comment (MariyaSin) [reply src/UserService.Validation/Gender/CreateGenderRequestValidator.cs:12]: remove trim

**Review by MariyaSin — APPROVED** [review]

**Review by actimellka — APPROVED** [review]

**Review by asyagraf — APPROVED** [review]

**Review by irlirla — APPROVED** [review]

**Pull request merged**